saleor/tests/fixtures.py

```diff
@@ -773,6 +773,22 @@ def color_attribute(db):  # pylint: disable=W0613
     return attribute
 
 
+@pytest.fixture
+def text_attribute(db):  # pylint: disable=W0613
+    attribute = Attribute.objects.create(
+        slug="text",
+        name="Text",
+        type=AttributeInputType.TEXT,
```

Comment:
`type` defines if this is `PRODUCT` or `PAGE` attribute, it means if the attribute can be assigned to `product_type` or to `page_type`. So here you should use both fields: `type` and `input_type`.

Also, some more tests should be added, at least:
- tests for updating/creating product type and (maybe also for page type) with text attributes
- test for assigning text attribute to product and variant